vulture_os/toolkit/api_parser/trendmicro_visionone/trendmicro_visionone.py

```diff
@@ -0,0 +1,180 @@
+
+Vulture OS is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Vulture OS.  If not, see http://www.gnu.org/licenses/.
+"""
+__author__ = "gparain"
+__credits__ = []
+__license__ = "GPLv3"
+__version__ = "4.0.0"
+__maintainer__ = "Vulture OS"
+__email__ = "contact@vultureproject.org"
+__doc__ = 'Trendmicro Visionone API Parser'
+__parser__ = 'TRENDMICRO VISIONONE'
+
+import json
+import logging
+import requests
+
+from datetime import datetime, timedelta
+from django.conf import settings
+from django.utils import dateparse, timezone
+
+from toolkit.api_parser.api_parser import ApiParser
+
+logging.config.dictConfig(settings.LOG_SETTINGS)
+logger = logging.getLogger('api_parser')
+
+
+
+class TrendmicroVisionOneAPIError(Exception):
+    pass
+
+
+class TrendmicroVisiononeParser(ApiParser):
+    URL_BASE = "https://api.eu.xdr.trendmicro.com"
+
+    def __init__(self, data):
+        """
+            trendmicro_visionone_token
+
+        """
+        super().__init__(data)
+
+        self.trendmicro_visionone_token = data["trendmicro_visionone_token"]
+        self.trendmicro_visionone_alerts_timestamp = data.get("trendmicro_visionone_alerts_timestamp")
+        self.trendmicro_visionone_audit_timestamp = data.get("trendmicro_visionone_audit_timestamp")
+        self.trendmicro_visionone_oat_timestamp = data.get("trendmicro_visionone_oat_timestamp")
+
+
+    def _get_alerts(self, since=None, to=None):
+
+        url_path = '/v3.0/workbench/alerts'
+
+        query_params = {'startDateTime': since,
+                        'endDateTime': to,
+                        'dateTimeTarget': 'createdDateTime',
+                        'orderBy': 'createdDateTime desc'}
+        headers = {'Authorization': 'Bearer ' + self.trendmicro_visionone_token, 'TMV1-Filter': ""}
+
+        result = requests.get(self.URL_BASE + url_path, params=query_params, headers=headers)
+        if result.status_code != 200:
+            raise TrendmicroVisionOneAPIError(
+                f"Error on URL: {self.URL_BASE + url_path} Status: {result.status_code} Reason/Content: {result.content}")
+
+        return result.json()['items']
+
+    def _get_auditlogs(self, since=None, to=None):
+        url_path = '/v3.0/audit/logs'
+
+        query_params = {'startDateTime': since,
+                        'endDateTime': to,
+                        'orderBy': 'createdDateTime desc',
+                        'top': 50,
+                        'labels': 'all'}
+        headers = {'Authorization': 'Bearer ' + self.trendmicro_visionone_token,
+                   'Accept': 'application/json',
+                   'TMV1-Filter': ''}
+
+        r = requests.get(self.URL_BASE + url_path, params=query_params, headers=headers)
```

Comment:
missing on every request :
- use of session to avoid having to provide headers (especially, to avoid forgetting them)
- use of proxy when available
- use of explicit timeouts to avoid overly long requests